### PR TITLE
Added Update operation for protection policy

### PR DESCRIPTION
**Pull request by prajwalpatil25**
# Description
Added Update functionality for protection policy.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

Tested execution of main.tf. Protection policy was updated successfully.
[Protection Policy Update operation testing.txt](https://github.com/dell/terraform-provider-powerstore/files/10302258/Protection.Policy.Update.operation.testing.txt)

### CODE REVIEW

**Review by mkmayank-dell — APPROVED** [review]
LGTM

File: powerstore/resource_protection_policy.go [hunk 1]
```diff
@@ -215,6 +215,58 @@ func (r resourceProtectionPolicy) Read(ctx context.Context, req tfsdk.ReadResour
...
+
+	//Get current state
+	var state models.ProtectionPolicy
+	diags = req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	protectionPolicyUpdate := r.planToProtectionPolicyParam(plan)
```
Comment (Sakshi-dell) [inline powerstore/resource_protection_policy.go:236]: Is removal of existing snapshot/replication rule possible with this?

Comment (shenda1) [inline powerstore/resource_protection_policy.go:236]: Please verify once if its possible to modify a single attribute.

Comment (prajwalpatil25) [reply powerstore/resource_protection_policy.go:236]: yes

Comment (prajwalpatil25) [reply powerstore/resource_protection_policy.go:236]: yes, single attribute can be modified.

**Review by Sakshi-dell — APPROVED** [review]

**Review by Sakshi-dell — APPROVED** [review]

**Pull request merged**